lab_3_generate_by_ngrams/main.py

```diff
@@ -159,6 +217,22 @@ def _postprocess_decoded_text(self, decoded_corpus: tuple[str, ...]) -> Optional
 
         In case of corrupt input arguments, None is returned
         """
+        if not isinstance(decoded_corpus, tuple) or len(decoded_corpus) < 1:
+            return None
+        string_text = ''
+        string_text = ''.join([string_text, decoded_corpus[0].upper()])
+        print(string_text)
+        for element in decoded_corpus[1:-1]:
+            if element.isalpha():
+                string_text = ''.join([string_text, element])
+            else:
+                string_text = ''.join([string_text, ' '])
```

Comment:
why not use `replace` once you have decoded everything?

Reply:
i rewrited, using replace once)